Additional packages to fix upgrading to RN 0.57

drewvolz opened the pull request:
Additional packages for upgrading to RN 0.57.0. Solution taken from [here](https://medium.com/@oleg2014/how-to-upgrade-project-to-react-native-0-57-rn-0-57-1a7e9fd8098).

* ~`@babel/plugin-external-helpers` @ v7.0.0~
* `metro-react-native-babel-preset` @ v0.48.1
  * The babel plugin was recently moved to the metro repo and published under metro-react-native-babel-preset
* ~`schedule` @ v0.4.0~

greenkeeper[bot] committed:
chore(package): update danger to version 4.4.8

Closes #3056

rye committed:
Merge branch 'master' into greenkeeper/danger-4.4.8

rye committed:
yarn: Fix the lockfile

[GitHub] Closes #3056.
[GitHub] Closes #3081.
[GitHub] Closes #3105.

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

hawkrives committed:
remove `import` statements

hawkrives committed:
enable DANGER_DISABLE_TRANSPILATION

hawkrives committed:
remove sudo:false from travis config

See: https://blog.travis-ci.com/2018-10-04-combining-linux-infrastructures

> Starting in the second week of November (we’ll post another blog post), we will start to migrate repositories that use sudo: false to the virtual machine-based infrastructure. Open source repositories will be rerouted first, with closed-source to follow. We’ll keep you posted via this blog and the changelog and twitter - have a look out for updates!

rye committed:
CONTRIBUTING: Refactor first and second paragraphs

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
CONTRIBUTING: Line wrap labels section

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
CONTRIBUTING: Line-wrap list in Labels section

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

drewvolz committed:
Additional packages to fix upgrading to RN 0.57

* @babel/plugin-external-helpers
* metro-react-native-babel-preset
* schedule

hawkrives reviewed (changes requested):
Please update your local copy of yarn to the latest version and regenerate the lockfile, so that you aren't removing all of the integrity information.

drewvolz committed:
regenerate lockfile for integrity

rye committed:
CONTRIBUTING: Line wrap each line in the contributing file

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
CONTRIBUTING: Also mention Circle

rye committed:
CONTRIBUTING: Dedent label list

rye committed:
CONTRIBUTING: Replace – with —

Our prior usage of the en-dash was incorrect.  Em-dash is correct in
this case.

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
CONTRIBUTING: Remove "CI notes"

We do not need this in this file.

I moved it to https://github.com/StoDevX/AAO-React-Native/wiki/CI-Environment-Variables.

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
CONTRIBUTING: Remove #ci-notes references

rye committed:
CONTRIBUTING: Balance whitespace up a bit

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
CONTRIBUTING: Nix a newline

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
CONTRIBUTING: Remove mention of Greenkeeper

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

hawkrives committed:
Merge pull request #3133 from StoDevX/remove-sudo-false-travis

remove sudo:false from travis config

rye committed:
Merge pull request #3135 from StoDevX/revamp-contributing

Revamp contributing documentation

rye committed:
Upgrade danger to version 4.4.10

Assuming nothing else is broken, this should actually be functional,
finally!

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
Merge pull request #3127 from StoDevX/greenkeeper/danger-4.4.8

Update Danger to version 4.4.8

rye committed:
Merge branch 'greenkeeper/react-native-0.57.0' into drew-greenkeeper/react-native-0.57.0

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

hawkrives committed:
fix "yarn run prettier"

It looks like Yarn changed their binary/script priority in favor of
binaries, which is a reversal of the status quo. IOW, `yarn prettier`
will now run `./node_modules/.bin/prettier` instead of the package.json
script "prettier".

Thus, I have changed the package.json script to "pretty".

Regarding the other change... For a number of months now, I've
been using "yarn p" as an alias for "yarn pretty-quick", and
I really like it, so I'm changing that.

hawkrives commented on package.json:
```diff
@@ -173,6 +174,7 @@
     "pretty-quick": "1.8.0",
     "prop-types": "15.6.2",
     "react-test-renderer": "16.5.0",
+    "schedule": "0.4.0",
```
@drewvolz What made us need this package? I don't recall reading about it being required…

rye replied:
It seems to be an internal dep of react. This may be my local environment polluting; I'm going to upgrade to the latest and see if that changes anything.

hawkrives replied:
React and RN are tied closely together; you ~~can't~~ shouldn't upgrade one without the other.

hawkrives replied:
RN 0.57.3 wants `react@16.6.0-alpha.8af6728`. 

(Also, RN 0.57.4 will be out in ~1 or 2 weeks, and will either go with 16.6.0-final or downgrade to 16.5.0.)

hawkrives replied:
I get no warnings from `yarn` when I remove it locally.

drewvolz replied:
@hawkrives 

1. `schedule@^0.3.0` is already included as it is a dependency of `react-test-renderer@16.5.0` ([see the yarn lockfile](https://github.com/StoDevX/AAO-React-Native/blob/17c6bcf185f0951cdc493f20efc17721c3884376/yarn.lock#L7135))
2. I've tried adding `schedule@0.4.0` due to advice in [here](https://github.com/facebook/react-native/issues/21150)

hawkrives committed:
fix dangerfile importing of yarn function

rye committed:
Merge pull request #3136 from StoDevX/fix-pretty-script

fix "yarn run prettier"

rye committed:
Merge branch 'master' into drew-greenkeeper/react-native-0.57.0

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

hawkrives committed:
fix delay dep in ccc-calendar module

hawkrives committed:
update react to requested peerDep version

hawkrives commented on package.json:
```diff
@@ -143,6 +144,7 @@
     "wordwrap": "1.0.0"
   },
   "devDependencies": {
+    "@babel/plugin-external-helpers": "7.0.0",
```
@drewvolz Why did we add this package, if our babel config is just `metro-react-native-babel-preset`?

hawkrives replied:
(I see that the article says to install it, but I don't like installing things that we don't directly use.)

drewvolz replied:
As suggested [here](https://github.com/facebook/react-native/issues/21219#issue-362115854) and fixed [here](https://github.com/facebook/metro/commit/48fa696ca6fc38b0bdd8c86feade620f3c4d0a34)

> Metro depends on a package that it does not specify in its package.json. It is causing the following error:
>
> Loading dependency graph, done.
error: bundling failed: Error: Cannot find module '@babel/plugin-external-helpers'
[...]

drewvolz replied:
But I haven't checked to see if it landed / we're picking it up.

hawkrives replied:
FWIW, https://github.com/facebook/metro/commit/48fa696ca6fc38b0bdd8c86feade620f3c4d0a34 says it landed in 0.45.5 and is still there in 0.48.1, according to the github tags list, and our Yarn lockfile / `npm info react-native` says that we're depending directly on metro@0.48.1 as of this branch.

Thus, I think we can remove this direct dependency.

drewvolz replied:
Great, thanks for looking.

drewvolz committed:
remove dep on @babel/plugin-external-helpers as it is provided by metro
